File: SDKs/bitbucket.org/2.0/python/src/sdk/models/operations/get_user_emails.py
from dataclasses import dataclass, field
from typing import Any,Optional


@dataclass
class GetUserEmailsSecurityOption1:
    oauth2: shared.SchemeOauth2 = field(default=None, metadata={'security': { 'scheme': True, 'type': 'oauth2' }})
    

@dataclass
class GetUserEmailsSecurityOption2:
    basic: shared.SchemeBasic = field(default=None, metadata={'security': { 'scheme': True, 'type': 'http', 'sub_type': 'basic' }})
    

@dataclass
class GetUserEmailsSecurityOption3:
    api_key: shared.SchemeAPIKey = field(default=None, metadata={'security': { 'scheme': True, 'type': 'apiKey', 'sub_type': 'header' }})
    

@dataclass
class GetUserEmailsSecurity:
    option1: Optional[GetUserEmailsSecurityOption1] = field(default=None, metadata={'security': { 'option': True }})
    option2: Optional[GetUserEmailsSecurityOption2] = field(default=None, metadata={'security': { 'option': True }})
    option3: Optional[GetUserEmailsSecurityOption3] = field(default=None, metadata={'security': { 'option': True }})
    

@dataclass
class GetUserEmailsRequest:
    security: GetUserEmailsSecurity = field(default=None)
    

@dataclass
class GetUserEmailsResponse:
    content_type: str = field(default=None)
    status_code: int = field(default=None)
    error: Optional[dict[str, Any]] = field(default=None)
    

File: SDKs/bitbucket.org/2.0/typescript/src/sdk/models/operations/getuseremails.ts
import { Metadata, SpeakeasyBase } from "../../../internal/utils/utils";
import * as shared from "../shared";


export class GetUserEmailsSecurityOption1 extends SpeakeasyBase {
  @Metadata({ data: "security, scheme=true;type=oauth2" })
  oauth2: shared.SchemeOauth2;
}


export class GetUserEmailsSecurityOption2 extends SpeakeasyBase {
  @Metadata({ data: "security, scheme=true;type=http;subtype=basic" })
  basic: shared.SchemeBasic;
}


export class GetUserEmailsSecurityOption3 extends SpeakeasyBase {
  @Metadata({ data: "security, scheme=true;type=apiKey;subtype=header" })
  apiKey: shared.SchemeApiKey;
}


export class GetUserEmailsSecurity extends SpeakeasyBase {
  @Metadata({ data: "security, option=true" })
  option1?: GetUserEmailsSecurityOption1;

  @Metadata({ data: "security, option=true" })
  option2?: GetUserEmailsSecurityOption2;

  @Metadata({ data: "security, option=true" })
  option3?: GetUserEmailsSecurityOption3;
}


export class GetUserEmailsRequest extends SpeakeasyBase {
  @Metadata()
  security: GetUserEmailsSecurity;
}


export class GetUserEmailsResponse extends SpeakeasyBase {
  @Metadata()
  contentType: string;

  @Metadata()
  statusCode: number;

  @Metadata()
  error?: Map<string, any>;
}
